imap-proto/src/parser/rfc3501.rs

```diff
@@ -727,5 +747,27 @@ mod tests {
             }
             rsp @ _ => panic!("unexpected response {:?}", rsp)
         }
+
+        match parse_response(b"* NO [BADCHARSET] error\r\n") {
+            Ok((_, Response::Data {
+                status: Status::No,
+                code: Some(ResponseCode::BadCharset(None)),
+                information: Some("error")
+            })) => {},
+            rsp @ _ => panic!("unexpected response {:?}", rsp)
+        }
+
+        match parse_response(b"* NO [BADCHARSET (utf-8 latin1)] error\r\n") {
```

Comment:
Would you mind adding a test here that has an empty list of charsets?

Reply:
Ok, I added it.